### ISSUE TITLE
Cannot suppress fn_to_numeric_cast_with_truncation

### ISSUE THREAD

**Issue by dtolnay** [rust-lang-nursery/rust-clippy#3276]
```rust
#![feature(tool_lints)]
#![allow(clippy::needless_pass_by_value)]
#![allow(clippy::fn_to_numeric_cast_with_truncation)]

pub fn f(s: String) {
    s.as_str();
}

fn main() {
    fn f() {}
    let _ = f as u8;
}
```

This code successfully suppresses the `needless_pass_by_value` lint. But it seems `fn_to_numeric_cast_with_truncation` cannot be suppressed and clippy produces the lint anyway.

```console
warning: casting function pointer `f` to `u8`, which truncates the value       
  --> src/main.rs:11:13                                                        
   |                                                                           
11 |     let _ = f as u8;                                                      
   |             ^^^^^^^ help: try: `f as usize`                               
   |                                                                           
   = note: #[warn(clippy::fn_to_numeric_cast_with_truncation)] on by default   
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#fn_to_numeric_cast_with_truncation
```

**Comment by phansch** [comment]
Thanks for the continued reports :heart:  Could you also provide the output of `cargo clippy -V`?

**Comment by dtolnay** [comment]
```console
$ cargo clippy -V
clippy 0.0.212 (32b1d1f 2018-10-05)
```
